src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -648,6 +679,12 @@ public OpenIdConnectHandler(HttpClient backchannel)
             ticket = userInformationReceivedContext.AuthenticationTicket;
             user = userInformationReceivedContext.User;
 
+            Options.ProtocolValidator.ValidateUserInfoResponse(new OpenIdConnectProtocolValidationContext()
+            {
+                ClientId = Options.ClientId,
+                ProtocolMessage = message,
```

Comment:
> 1. UserInfoEndpointResponse: which is the user info endpoint response and of type JwtSecurityToken. The endpoint response will become the JwtPayload of the token.

Interesting. Curious: when the userinfo response is not signed nor encrypted, a simple JSON object is returned and you have to manually "create" a JWT for `ValidateUserInfoResponse` to work correctly. Have you considered accepting a `JObject`?